mlkem/poly.h

```diff
@@ -238,8 +239,39 @@ void poly_getnoise_eta1122_4x(poly *r0, poly *r1, poly *r2, poly *r3,
 
 #define poly_basemul_montgomery_cached \
   MLKEM_NAMESPACE(poly_basemul_montgomery_cached)
+/*************************************************
+ * Name:        poly_basemul_montgomery_cached
+ *
+ * Description: Multiplication of two polynomials in NTT domain,
+ *              using mulcache for second operand.
+ *
+ *              Bounds:
+ *              - a is assumed to be coefficient-wise < q in absolute value.
+ *              - b is assumed to be the output of a forward NTT and
+ *                thus coefficient-wise bound by C
+ *              - b_cache is assumed to be coefficient-wise bound by D.
```

Comment:
I am confused by these bounds. `basemul_cached` says b and b_cache can be arbitrary int16 values. Should these bounds be relaxed here?
If I do understand correctly, then the b is bouned by `NTT_BOUND` (`8 * MLKEM_Q`) and `b_cache` does not have a bound that is guaranteed (although it likely has the same bounds as fqmul)?

Reply:
Good spot, removed this part of the doc to make it uniform with the formal spec